contracts/base/TokenCollateralLoans.sol

```diff
@@ -147,6 +147,24 @@ contract TokenCollateralLoans is LoansBase {
         collateralTokenTransferFrom(msg.sender, amount);
     }
 
+    function _requireExpectedBalance(uint256 initialBalance, uint256 expectedAmount)
+        internal
+        view
+    {
+        uint256 finalBalance = ERC20Detailed(collateralToken).balanceOf(address(this));
+        if (finalBalance > initialBalance) {
```

Comment:
I think it would be better to specify an additional function parameter indicating the direction of transfer, and use that for the assertions. Using the condition `if (finalBalance > initialBalance)` reads a little weird to me.

Reply:
Fixed, now it receives isTransfer param. Also, I added some unit tests.